tests/unit/test_network_connections.py

```diff
@@ -3670,6 +3707,95 @@ def test_path_to_glib_bytes(self):
         self.assertEqual(result.get_data(), b"file:///my/test/path\x00")
 
 
+class TestValidatorMatch(unittest.TestCase):
+    def setUp(self):
+        self.test_profile = {
+            "name": "test",
+            "type": "ethernet",
+            "ip": {
+                "dhcp4": False,
+                "address": "192.168.245.7/24",
+            },
+            "match": None,
+        }
+        self.validator = network_lsr.argument_validator.ArgValidator_DictConnection()
+
+    def test_match_path_empty_list(self):
+        """
+        I can remove a match path setting by setting it to empty list.
+        """
+        self.test_profile["match"] = {"path": []}
+        result = self.validator.validate(self.test_profile)
+        self.assertEqual(result["match"], {"path": []})
+
+    def test_match_path_empty_string_in_list(self):
+        """
+        Test that [""] is allowed for 'match.path' setting
+        """
+        self.test_profile["match"] = {"path": [""]}
+        result = self.validator.validate(self.test_profile)
+        self.assertEqual(result["match"], {"path": [""]})
+
+    def test_match_path_valid_setting(self):
+        """
+        Test that values like ["&"] and ["pci-0000:00:03.0"] are valid values for
+        'match.path' setting
+        """
+        self.test_profile["match"] = {"path": ["&"]}
+        result1 = self.validator.validate(self.test_profile)
+        self.assertEqual(result1["match"], {"path": ["&"]})
+
+        self.test_profile["match"] = {"path": ["pci-0000:00:03.0"]}
+        result2 = self.validator.validate(self.test_profile)
+        self.assertEqual(result2["match"], {"path": ["pci-0000:00:03.0"]})
+
+    def test_match_path_invalid_connection_type(self):
+        """
+        Test that when 'match.path' setting is correctly defined but the connection
+        type is neither ethernet nor infiniband, a ValidationError raises.
+        """
+        self.test_profile["match"] = {"path": ["pci-0000:00:03.0"]}
+        result = self.validator.validate(self.test_profile)
+        self.assertEqual(result["match"], {"path": ["pci-0000:00:03.0"]})
+
+        self.test_profile["type"] = "dummy"
+        self.assertRaises(ValidationError, self.validator.validate, self.test_profile)
```

Comment:
Please use [asssertRaisesRegexp](https://docs.python.org/2.7/library/unittest.html#unittest.TestCase.assertRaisesRegexp) to match the specific message of the exception to ensure that it is the one that we are expecting to avoid surprising messages as in https://github.com/linux-system-roles/network/pull/363#issuecomment-875811247

Reply:
I used `asssertRaisesRegexp` instead of `assertRaises`